test/CMakeLists.txt

```diff
@@ -70,6 +70,38 @@ if(AF_BUILD_UNIFIED)
   list(APPEND enabled_backends "unified")
 endif(AF_BUILD_UNIFIED)
 
+
+add_library(arrayfire_test STATIC
+  testHelpers.hpp
+  arrayfire_test.cpp)
+
+target_include_directories(arrayfire_test
+  PRIVATE . ../include ../build/include ../extern/half/include)
+
+target_link_libraries(arrayfire_test
+  PRIVATE
+    gtest_main
+  PUBLIC
+    gtest
+  )
+
+if(AF_TEST_WITH_MTX_FILES)
+  target_link_libraries(arrayfire_test PRIVATE mmio)
```

Comment:
Shouldn't this be public ? I mean you can perhaps remove this from linking command in `make_test` macro and make it PUBLIC dependency here.